src/main/native/org/apache/commons/crypto/OpenSslInfoNative.c

```diff
@@ -135,3 +96,35 @@ JNIEXPORT jstring JNICALL Java_org_apache_commons_crypto_OpenSslInfoNative_Nativ
 {
     return (*env)->NewStringUTF(env, PROJECT_NAME);
 }
+
+JNIEXPORT jlong JNICALL Java_org_apache_commons_crypto_OpenSslInfoNative_OpenSSL
+  (JNIEnv *env, jclass clazz)
+{
+    if (!load_library(env)) {
+        return 0;
+    }
+    if(OPENSSL_VERSION_NUMBER > VERSION_1_1_X){
```

Comment:
nit: space after `if` (also in some other places)

Where does `OPENSSL_VERSION_NUMBER` come from? A quick google search seems to indicate it's a macro in an OpenSSL header. Which makes this a compile time check, basically.

Reply:
I replaced it with a dynamic lookup, following the existing pattern of dlsyming functions via a load macro.